server/src/main/java/org/opensearch/action/admin/cluster/stats/ClusterStatsIndices.java

```diff
@@ -202,6 +223,12 @@ public static class ShardStats implements ToXContentFragment {
 
         public ShardStats() {}
 
+        public ShardStats(StreamInput in) throws IOException {
```

Comment:
This was an internal object in ClusterStatsIndices, why are adding serialization to it.

Reply:
This is done since we are now precomputing and storing the node level aggregated stats in the form of these ShardStats. 

This table can explain the pre-computation - 

| Before | After |
| --- | --- | 
| On each node, `indices.stats.ShardStats` is fetched from the indices service which contains stats of each shard on the node | On each node, ShardStats is fetched from the indices service which contains stats of each shard on the node |
| We return this list of `indices.stats.ShardStats` to the coordinator node | We iterate over the `indices.stats.ShardStats` returned from IndicesService and create `ClusterIndicesStats.ShardStats` for that specific node. We now create a hashmap of index_name to `ClusterIndicesStats.ShardStats` at a node level. This is then returned to the coordinator node |
| Coordinator node iterates over each `indices.stats.ShardStats` from each node response and creates a map of index_name to `ClusterIndicesStats.ShardStats`, i.e., for each index, the `indices.stats.ShardStats` returned from the nodes are iterated and added into the respective `ClusterIndicesStats.ShardStats`. In the end, the `countsPerIndex` map contains indices and their accumulated `ClusterIndicesStats.ShardStats` made by adding up values from the `indices.stats.ShardStats` from each node | The coordinator node iterates over each Node response and merges the `countsPerIndex` map from each node to make a final stats result. |


Since we are now creating `ClusterIndicesStats.ShardStats` on each node, we require this to be capable of serialising and deserialising.